enhancements/unmanaged-clair-database.md

```diff
@@ -0,0 +1,142 @@
+
+```
+apiVersion: quay.redhat.com/v1
+kind: QuayRegistry
+metadata:
+  name: quay
+spec:
+  components:
+  - kind: clairdatabase
+    managed: false
+  configBundleSecret: config-bundle-secret
+
+```
+
+The provided clair-config.yaml would be mounted into the Clair pod, and any fields not provided should be automatically populated with defaults using the Clair config module.
+
+### Implementation Details/Notes/Constraints
+
+One implementation detail to note is that a managed Clair component should consume a custom `clair-config.yaml` as long as it is provided. This is similar to how the base Quay component can be customized with a `config.yaml` despite always being a managed component. However, the clair config must be in sync with the state of the `clairdatabase` component. For example, if a user chooses a managed `clairdatabase` component, the `clair-config.yaml` should not contain custom DSN values.
+
+### Risks and Mitigations
+
+Allowing for further configuration increases the likelihood of incorrect configurations. The operator must ensure that the configuration is in fact valid, and report any incorrect configurations. To mitigate this, the Quay Operator documentation should provide a reference to the Clair documentation on how to set up a database and attach this to the `clair-config.yaml`.
+
+## Design Details
+
+### Test Plan
+
+```
+| `clair` | `clairdatabase` | Clair Config with DSN Provided | Expected result |
+|---|---|---|---|
+| Managed | Managed | Yes | Error, managed `clairdatabase` should not contain a DSN |
+| Managed | Managed | No | No Error, this is a fully managed Clair instance |
+| Managed | Unmanaged | No | Error, Clair config must contain DSN |
+| Managed | Unmanaged | Yes | No Error, Clair will use the DSN provided in the custom config |
+| Unmanaged | Unmanaged | No | Error, unmanaged Clair should not have a clair config |
+| Unmanaged | Unmanaged | Yes | Error, unmanaged Clair should not have a clair config |
+| Unmanaged | Managed | No | Error, unmanaged Clair should not have a clair config |
+| Unmanaged | Managed | Yes | Error, unmanaged Clair should not have a clair config |
+```
+
+This matrix can be tested using unit tests.
+
+In addition to this, e2e tests should be implemented inside a real cluster. This can be done using the `kuttl` tool. A PR has been opened to add the `kuttl` e2e tests to the operator and can be found at https://github.com/quay/quay-operator/tree/kuttl
+
+### Upgrade / Downgrade Strategy
+
+Since previous versions of Quay do not contain a `clair-config.yaml` in the `config-bundle-secret`, and do not have a customizable `clairdatabase` component, upgrades should not affect the behavior of any `QuayRegistry`. After an upgrade, the operator should automatically set `clairdatabase` to be a managed component, and the default generation of the clair config should proceed as normal.
```

Comment:
Small consideration here, when you say: "After an upgrade, the operator should automatically set `clairdatabase` to be a managed component" I would add "if Clair is managed". If during the upgrade clair is not managed then we don't need to spawn its database. wdyt?

Reply:
Good point, I will add this